mongo/src/org/immutables/mongo/repository/RepositorySetup.java

```diff
@@ -136,7 +182,37 @@ public Builder database(MongoDatabase database) {
      * @return {@code this}
      */
     public Builder gson(Gson gson) {
-      this.gson = checkNotNull(gson);
+      checkNotNull(gson);
+      this.codecRegistry = CodecRegistries.fromRegistries(MongoClient.getDefaultCodecRegistry(),
+              Codecs.codecRegistryFromGson(gson)
+              );
+
+      // legacy naming
+      this.fieldNamingStrategy = new FieldNamingStrategy.GsonNamingStrategy(gson);
+      return this;
+    }
+
+    /**
+     * Register custom codec registry to serialize entities externally.
+     *
+     * @param registry existing registry
+     * @return {@code this}
+     */
+    Builder codecRegistry(CodecRegistry registry) {
+      this.codecRegistry = checkNotNull(registry, "registry");
+      this.fieldNamingStrategy = null;
+      return this;
+    }
+
+    /**
+     * Register custom property naming strategy. Allows to rename POJO properties
+     * during query construction.
+     *
+     * @param fieldNamingStrategy instance to be used during querying
+     * @return {@code this}
+     */
+    Builder fieldNamingStrategy(@Nullable FieldNamingStrategy fieldNamingStrategy) {
```

Comment:
either `@Nullable` is accidental or I don't understand )

Reply:
It is accidental